packages/ove-app-maps/docs/MAP_LAYERS_JSON.md

```diff
@@ -0,0 +1,51 @@
+# MapLayers.json File
+
+The maps app can be configured by embedding the map layers configuration within the `config.json` file, or by providing a URL that points to a `MapLayers.json` file. The format of the JSON configuration remains the same in either of these two approaches.
+
+The `MapLayers.json` file contains an array of map layers. It has a structure similar to:
+
+```JSON
+[
+    {
+        "type": "ol.layer.Tile",
+        "visible": false,
+        "wms": false,
+        "source": {
+            "type": "ol.source.OSM",
+            "config": {
+                "crossOrigin": null,
+                "url": "https://{a-c}.tile.thunderforest.com/transport/{z}/{x}/{y}.png"
+            }
+        }
+    },
+    {
+        "type": "ol.layer.Vector",
+        "visible": false,
+        "wms": true,
+        "source": {
+            "type": "ol.source.Vector",
+            "config": {
+                "url": "/data/sampleGeo.json"
+            }
+        },
+        "style": {
+            "fill": {
+                "color": "#B29255"
+            },
+            "stroke": {
+                "color": "#715E3A",
+                "width": 4
+            }
+        },
+        "opacity": 0.7
+    }
+]
+```
+
+The app supports two types of map layers, `ol.layer.Tile` and `ol.layer.Vector`.
+
+Both types of map layers have a `visible` parameter that describes whether the layer is visible by default and a `wms` parameter which describes whether the layer is a displaying data from a [`Web Map Service`](https://www.opengeospatial.org/standards/wms). These parameters have boolean values.
+
+The two types of layers also have a common `source` parameter, but their contents differ. The `source` of a vector layer always must have a `type` equal to `ol.source.Vector`, and the `url` of the `config` parameter must point to a [GeoJSON](http://geojson.org/) file. The `source` of a tile layer can be any type of source providing maps as tiled images such as `ol.source.OSM` or `ol.source.BingMaps`. The contents of `config` changes according to the source and more information can be found in the [OpenLayers documentation](http://geoadmin.github.io/ol3/apidoc/ol.source.html).
```

Comment:
1.  ~~The two types of layers also~~ **Both types of layer** have a common `source` parameter
2. The `source` of a tile layer can be any ~~type of source providing maps as tiled images~~ [map layer source supported by OpenLayers 3](https://geoadmin.github.io/ol3/apidoc/ol.source.html).
3. The `config` object is passed as an argument of the constructor for the specified source.

**The `config` object is passed as an argument to the constructor for the specified source** ~~The contents of `config` changes according to~~**, and its properties depend on which source is used;** ~~and~~ more information can be found in the [OpenLayers documentation](http://geoadmin.github.io/ol3/apidoc/ol.source.html).

Reply:
It should be **map ~~layer~~ tile source supported by OpenLayers ~~3~~**.

Reply:
The docs refer to "layer source" rather than "tile source" (e.g., [here](https://geoadmin.github.io/ol3/apidoc/ol.source.BingMaps.html): "Layer source for Bing Maps tile data", or "Layer source for the MapQuest tile server").

Also, the supported sources differ between OpenLayers versions (e.g., CartoDB [is a source in v5.3.0](https://openlayers.org/en/latest/apidoc/module-ol_source_CartoDB.html), but not in OpenLayers 3).

Reply:
@jamesscottbrown, I can fix to OpenLayers 3, but I dropped it since we would be migrating to a newer version of OL and that whole reference doesn't make sense at that point. When OpenLayers was version 3, people used to call it "OpenLayers 3", but beyond that version number, be it 4.x.y or 5.x.y or 6.x.y, they call it "Open Layers" (w/o version number).

I think we were both wrong on "tile source" vs "layer source" (slightly better than the other). It should be "layer source for tile data"

Reply:
Ok - and I suppose the ambiguity is reduced because we link to the API docs for version 3.